### PR TITLE
Minor POM cleanup

### PR DESCRIPTION
**Pull request by basil**
Cleans up the build in various minor ways:

- Removes the unused `animal-sniffer-annotations` dependency
- Moves the `test-annotations` from the root Maven project into the `core/` and `test/` subprojects (the only two where it is actually used), and removes the version since it is defined in the grandparent POM
- Removes `org.objenesis` from the `<dependencyManagement>` section because it doesn't need to be there
- Define the `mockito-inline` version as a simple property rather than in the `<dependencyManagement>` for consistency

### Proposed changelog entries

N/A

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [ ] (If applicable) Jira issue is well described
- [ ] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [ ] Appropriate autotests or explanation to why this change has no tests
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

**Comment by timja** [comment]
```

[2021-11-29T20:40:22.251Z] [ERROR] The build could not read 2 projects -> [Help 1]

[2021-11-29T20:40:22.251Z] org.apache.maven.project.ProjectBuildingException: Some problems were encountered while processing the POMs:

[2021-11-29T20:40:22.251Z] [ERROR] 'dependencies.dependency.version' for org.mockito:mockito-inline:jar must be a valid version but is '${mockito.version}'. @ org.jenkins-ci.main:jenkins-core:${revision}${changelist}, /home/jenkins/workspace/Core_jenkins_PR-5973/core/pom.xml, line 415, column 16

[2021-11-29T20:40:22.251Z] [ERROR] 'dependencies.dependency.version' for org.mockito:mockito-inline:jar must be a valid version but is '${mockito.version}'. @ org.jenkins-ci.main:jenkins-test:${revision}${changelist}, /home/jenkins/workspace/Core_jenkins_PR-5973/test/pom.xml, line 162, column 16
```

**Comment by basil** [comment]
Bad merge, fixed

**Review by timja — APPROVED** [review]

**Review by res0nance — APPROVED** [review]

**Comment by timja** [comment]
This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Pull request merged**
